tools/common_lib/src/gemm.h

```diff
@@ -351,6 +354,108 @@ class Gemm : public DirectMlBaseNode
             auto gemm_inp_b = dml::Reinterpret(reshaped_split, dml::TensorDimensions{ batch, head_count, seq, head_size }, input_1_strides);
             outputs_[0] = dml::Gemm(input_0_, gemm_inp_b, dml::NullOpt, DML_MATRIX_TRANSFORM_NONE, DML_MATRIX_TRANSFORM_NONE, alpha, beta, fused_act);
         }
+        else if (type_ == GemmType::GemmType_Quant)
```

Comment:
As written above - if you extend this class than no need for another type, and `GemmType::GemmType_AB` condition could be modified.

Reply:
Have restructured the code and made Quant Gemm to be a separate class.